spinalcordtoolbox/registration/core.py

```diff
@@ -424,17 +430,11 @@ def register(src, dest, step, param):
                '\nERROR: ANTs failed. Exit program.\n', 1, 'error')
     else:
         # rename warping fields
-        if (step.algo.lower() in ['rigid', 'affine', 'translation'] and
-                step.slicewise == '0'):
-            # if ANTs is used with affine/rigid --> outputs .mat file
-            warp_forward = 'warp_forward_' + str(step.step) + '.mat'
-            os.rename(warp_forward_out, warp_forward)
-            warp_inverse = '-warp_forward_' + str(step.step) + '.mat'
-        elif step.type in ['label']:
-            # if label-based registration is used --> outputs .txt file
-            warp_forward = 'warp_forward_' + str(step.step) + '.txt'
+        _, _, output_ext = extract_fname(warp_forward_out)
+        if output_ext in ['.txt', '.mat']:
+            warp_forward = 'warp_forward_' + str(step.step) + output_ext
             os.rename(warp_forward_out, warp_forward)
-            warp_inverse = '-warp_forward_' + str(step.step) + '.txt'
+            warp_inverse = '-warp_forward_' + str(step.step) + output_ext
         else:
             warp_forward = 'warp_forward_' + str(step.step) + '.nii.gz'
             warp_inverse = 'warp_inverse_' + str(step.step) + '.nii.gz'
```

Comment:
I've added extended commit messages to my commits, but there's one snippet I wrote that I want to explicitly want to highlight here:

> That said, this section of code still gives me pause:
> 
> - Why is it that the 'else:' case renames both warp_forward and warp_inverse, but the 'if:' case only renames 'warp_forward'?
> - In that case, why even create the 'warp_inverse' variable at all?
> - Does the 'warp_inverse' file just not exist?
> - If we know that the 'warp_inverse' file doesn't get created in the '.mat' and '.txt' cases, then why not return e.g. 'None' for such a filename?
> 
> This requires a deeper dive into how the registration commands determine their output files, though, and I don't want to balloon the scope of this PR too much.

I'm not sure if now's the best time to refactor this section of code, but it definitely confused me / set off red flags when I first tried to read through it.

Reply:
These are great questions, but they would take more digging than we probably want to do in this PR, as you said.